pkg/generate/output/artifact.go

```diff
@@ -0,0 +1,178 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package output
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/aws-controllers-k8s/code-generator/pkg/version"
+)
+
+const (
+	outputFileName = "output.yaml"
+)
+
+// UpdateReason is the reason a package got modified.
+type UpdateReason string
+
+const (
+	// UpdateReasonAPIGeneration should be used when an API package
+	// is modified by the APIs generator (ack-generate apis).
+	UpdateReasonAPIGeneration UpdateReason = "API generation"
+
+	// UpdateReasonConversionFunctionsGeneration Should be used when
+	// an API package is modified by conversion functions generator.
+	// TODO(hilalymh) ack-generate conversion-functions
+	UpdateReasonConversionFunctionsGeneration UpdateReason = "Conversion functions generation"
+)
+
+// Artifact represent the parameters used the generate/update the
+// API version directory.
+//
+// This type is public because soon it will be used by conversion generators
+// to load APIs artifacts.
+// TODO(hilalymh) Add functions to load/edit artifact files.
+type Artifact struct {
+	// The APIs version e.g v1alpha2
+	ApisVersion string `json:"apis_version"`
+	// The checksum of all the combined files generated within the APIs directory
+	ApisDirectoryChecksum string `json:"apis_directory_checksum"`
```

Comment:
I would be fine with us referring to the plural of "API" as "API"? At least lets capitalise "Apis" to "APIs".

Reply:
++, surprisingly `go-lint` didn't complain about this..

Reply:
the `command` package is also using plural of API that why i went with this

Reply:
Changed to `APIVersion` which totally makes sense for this package